akvo/rsr/static/scripts-src/my-results/components/common.jsx

```diff
@@ -169,3 +174,65 @@ export class FileReaderInput extends React.Component {
     </div>
   }
 }
+
+// Markdown Editor to be used in the results framework
+export class MarkdownEditor extends React.Component {
+    constructor(props) {
+        super(props);
+        const {text, show_preview} = this.props;
+        this.state = {
+            reactMde: {text, selection: null},
+            show_editor: !show_preview,
+        };
+        this.showEditor = this.showEditor.bind(this);
+        this.showPreview = this.showPreview.bind(this);
+        this.onChange = this.onChange.bind(this);
+    }
+
+    showEditor() {
+        this.setState({show_editor: true});
+    }
+
+    showPreview() {
+        this.setState({show_editor: false});
+    }
+
+    onChange (reactMde) {
+        const {text, selection} = reactMde;
+        this.setState({reactMde})
+        this.props.onChange(text, selection);
+    }
+
+    render() {
+        const {show_editor} = this.state;
+        const editClass = show_editor?'active': ''
```

Comment:
Here's a weird one. If I append ";" to this line and the next, as my editor tells me to, the update form preview is always visible. I'm too tired to figure out why now, but maybe you can tomorrow? (If you can reproduce the behaviour...)

![screenshot_231](https://user-images.githubusercontent.com/142613/34693271-3d5a33a2-f4c3-11e7-8e87-83161b74e616.png)

Reply:
I wasn't able to reproduce this. The `editClass` is only used to change how the 'Write' and 'Preview' buttons look, not to control whether preview button is shown or not. 

Can you try running `npm install` before running `npm run watch`?